app/Http/Livewire/Components/Market/MarketComponent.php

```diff
@@ -0,0 +1,209 @@
+
+namespace App\Http\Livewire\Components\Market;
+
+use App\Models\Trade;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\DB;
+use Livewire\Component;
+
+class MarketComponent extends Component
+{
+    public $isCreateModalOpen = false;
+    public $isRespondModalOpen = false;
+
+    public $trades;
+    public $trade;
+
+    public $paginator = [];
+    public $page = 1;
+    public $itemsPerPage = 15;
+    public $totalItems = 0;
+
+    public $filter = [
+        'hydrogen_type'     => ['green', 'blue', 'grey', 'mix'],
+        'units_per_hour'    => [0, 0],
+        'duration'          => [0, 0],
+        'total_volume'      => [0, 0],
+        'price_per_unit'    => [0, 0],
+        'mix_co2'           => [0, 0],
+        'trade_type'        => ['offer', 'request'],
+    ];
+
+    public $sort = [
+        'hydrogen_type'     => ['Hydrogen type', ''],
+        'units_per_hour'    => ['Units per hour', ''],
+        'duration'          => ['Duration', ''],
+        'total_volume'      => ['Total volume', ''],
+        'price_per_unit'    => ['Price per unit', ''],
+        'mix_co2'           => ['Mix % CO2', ''],
+        'trade_type'        => ['Trade type', ''],
+    ];
+
+    protected $listeners = ['listingCreated' => 'listingCreated'];
+
+    public function mount()
+    {
+        $this->updateTrades(true);
+    }
+
+    public function render()
+    {
+        return view('livewire.components.market.market-component')->extends('layouts.app');
+    }
+
+    public function toggleCreateModal()
+    {
+        $this->isCreateModalOpen = !$this->isCreateModalOpen;
+    }
+
+    public function openRespondModal(Trade $trade)
+    {
+        $this->trade = $trade;
+        $this->isRespondModalOpen = true;
+        $this->emit('listingSelected', $trade);
+    }
+
+    public function closeRespondModal()
+    {
+        $this->isRespondModalOpen = false;
+    }
+
+    public function listingCreated()
+    {
+        // Close create modal
+        $this->toggleCreateModal();
+
+        // Update trades since we added a new one
+        $this->updateTrades();
+    }
+
+    public function updateTrades($setup = false)
+    {
+        // Determine the bounds and filters
+        if ($setup)
+        {
+            $this->determineStartingFilters();
+        }
+
+        // Apply filters and pagination
+        $trades = $this->getListings();
+
+        $trades = $trades->paginate($this->itemsPerPage, ['*'], 'page', $this->page);
+
+        $this->totalItems = $trades->total();
+        $this->paginator = $trades->toArray();
+        $this->trades = $trades->items();
+    }
+
+    private function determineStartingFilters()
+    {
+        // Determine bounds for database fields
+        foreach ($this->filter as $key => $value) {
+            // Skip calculated and enum fields
+            if ($key == 'hydrogen_type' || $key == 'trade_type' || $key == 'total_volume')
+            {
+                continue;
+            }
```

Comment:
`// Determine bounds for database fields
        foreach (collect($this->filter)->except(['hydrogen_type','trade_type','total_volume']) as $key => $value) {
            $this->filter[$key] = [Trade::min($key), Trade::max($key)];
        }`

could also work. Then you can keep the DoNotValidate fields in a property on the component, or even the model and loop through that!